### DISCUSSION TITLE
How do you compile a standalone executable?

### DISCUSSION THREAD

**Discussion by SmartBoy84** [airsquared/blobsaver#329] [Q&A] [answered]
Hey,

I've been trying to compile TSSChecker for the last couple of hours for the PI. The compilation went smoothly but depended on dynamic libraries which had to be present for the binary to load.

Yet, the TSSchecker binaries that you package are not dynamic executables.
How did you manage this? Did you compile each library listed as a dependency on the project's page individually then link it statically?
(I took a shortcut by just installing deb's of the libraries I found online 😛 - Do I seriously need to manually compile each library?)

I assume you added the '-static' flag, what did you do to get GCC to statically link the libraries though?
It would be **amazing** if you could compile this for me (but, if not, then guidance would be just as nice)!

**Comment by airsquared** [comment]
A dynamic executable should still work for blobsaver, you shouldn't need a standalone one.

If you really do want a standalone executable you will need to compile the dependencies yourself (although I don't recommend this for the time it takes). This shouldn't be too bad, just remember when compiling each library to pass `--enable-static --disable-shared`.

**Comment by SmartBoy84** [comment]
I actually just wanted to compile a standalone ececutable so I could have a lightweight method to save blobs regularly on my pi.

What libraries did you manually compile?

**Comment by airsquared** [comment]
I manually compiled everything that was in the dependencies list for tsschecker.

**Comment by SmartBoy84** [comment]
So, you compiled every library with the gcc argument "--enable-state --disable-shared"
What did you change to get the TSSChecker to use the static libraries?

**Comment by airsquared** [comment]
Not gcc arguments, `--enable-static --disable-shared` are arguments to `./autogen.sh` or `./configure`. I think I also might have added something like `-static` to LDFLAGS when compiling tsschecker itself, but I don't remember for sure.

**Comment by SmartBoy84** [comment]
Well, that didn't work.

I'm currently getting the following errors when running make for tsschecker:
```
usr/bin/ld: /usr/local/lib/libfragmentzip.so: undefined reference to symbol 'EVP_PKEY_id@@OPENSSL_1_1_0'
/usr/bin/ld: //lib/arm-linux-gnueabihf/libcrypto.so.1.1: error adding symbols: DSO missing from command line
```

**Comment by airsquared** [comment, marked as answer]
If you are still trying to do this (sorry I've been pretty busy recently), I found [these builds](https://github.com/1Conan/tsschecker/releases) that might work for you.

**Reply by SmartBoy84** [reply]
Thanks!
